### PR TITLE
fix: unmatched type for value in oneOf should be ignored

### PR DESCRIPTION
**Pull request by joaopedrocampos**
#### Checklist
- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [x] commit message and code follows the [Developer’s Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11) and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

Related issue: https://github.com/fastify/fast-json-stringify/issues/342

### CODE REVIEW

File: test/oneof.test.js [hunk 1]
```diff
@@ -401,3 +401,105 @@ test('oneOf object with field of type string with format or null', (t) => {
     prop: toStringify
   }), `{"prop":"${toStringify.toISOString()}"}`)
 })
+
+test('one array item do not match oneOf types', (t) => {
```
Comment (mcollina) [inline test/oneof.test.js:405]: I think this test title should be positive, not negative.

File: test/oneof.test.js [hunk 2]
```diff
@@ -401,3 +401,105 @@ test('oneOf object with field of type string with format or null', (t) => {
...
+  const stringify = build(schema)
+
+  const responseWithMappedType = stringify({
+    data: [false, 'foo']
+  })
+
+  t.equal('{"data":["foo"]}', responseWithMappedType)
+})
+
+test('some array items does not match oneOf types', (t) => {
```
Comment (mcollina) [inline test/oneof.test.js:439]: I think this test title should be positive, not negative.

File: index.js [hunk 3]
```diff
@@ -1023,9 +1023,11 @@ function buildArray (location, code, name, key = null) {
 
   code += `
     var l = obj.length
-    var w = l - 1
+
     for (var i = 0; i < l; i++) {
-      if (i > 0) {
+      var jsonLastChar = json.slice(-1)
+
+      if (i > 0 && jsonLastChar != '[' && jsonLastChar != ',') {
```
Comment (mcollina) [inline index.js:1030]: please use `!==`

File: index.js [hunk 4]
```diff
@@ -1023,9 +1023,11 @@ function buildArray (location, code, name, key = null) {
 
   code += `
     var l = obj.length
-    var w = l - 1
+
     for (var i = 0; i < l; i++) {
-      if (i > 0) {
+      var jsonLastChar = json.slice(-1)
```
Comment (mcollina) [inline index.js:1028]: I would do: `json[json.length - 1]`.

**Review by mcollina — CHANGES_REQUESTED** [review]

File: index.js (hunk 3 above)
Comment (joaopedrocampos) [reply index.js:1030]: My bad, totally missed this, gonna fix it 🙏

File: index.js (hunk 4 above)
Comment (joaopedrocampos) [reply index.js:1028]: Sure, gonna change to this

File: test/oneof.test.js (hunk 2 above)
Comment (joaopedrocampos) [reply test/oneof.test.js:439]: It really should be positive, my 🇺🇸 that tricked me

File: test/oneof.test.js (hunk 1 above)
Comment (joaopedrocampos) [reply test/oneof.test.js:405]: It really should be positive, my 🇺🇸 that tricked me

**Review by mcollina — APPROVED** [review]
lgtm

**Pull request merged**